include/ola/rdm/RDMCommand.h

```diff
@@ -49,199 +50,163 @@ namespace rdm {
+  /**
+   * @brief A set of values representing CommandClasses in E1.20.
+   * @note Please see section 6.2.10 of ANSI E1.20 for more information.
+   */
+  // TODO(simon): remove this in favor of the new one in RDMEnums.h
+  typedef enum {
+    DISCOVER_COMMAND = 0x10, /**< Discovery Command */
+    DISCOVER_COMMAND_RESPONSE = 0x11, /**< Discovery Response */
+    GET_COMMAND = 0x20, /**< Get Command */
+    GET_COMMAND_RESPONSE = 0x21, /**< Get Response */
+    SET_COMMAND = 0x30, /**< Set Command */
+    SET_COMMAND_RESPONSE = 0x31, /**< Set Response */
+    INVALID_COMMAND = 0xff, /**< Invalid Command, specific to OLA */
+  } RDMCommandClass;
+
+  virtual ~RDMCommand();
+
+  /**
+   * @name String Methods
+   * @{
+   */
+
+  /**
+   * @brief Create a human readable string from the RDMCommand object.
+   * @returns A string containing the source and destination UIDs, transaction
+   * number, port ID, Message count, Sub Device, Cmd Class, Param ID, Data,
+   * and a raw string of the parameter data.
+   */
+  std::string ToString() const;
+
+  /**
+   * @brief Output an RDMCommand object to an ostream.
+   * @param out ostream to output to
+   * @param command is the RDMCommand to print
+   * @sa ToString()
+   */
+  friend std::ostream& operator<<(std::ostream &out,
+                                  const RDMCommand &command) {
+    return out << command.ToString();
+  }
+
+  /** @} */
+
+  /**
+   * @brief The CommmandClass of this message.
+   */
+  virtual RDMCommandClass CommandClass() const = 0;
+
+  /**
+   * @name Accessors
+   * @{
+   */
+
+  /** @brief Returns the Source UID of the RDMCommand */
+  const UID& SourceUID() const { return m_source; }
+
+  /** @brief Returns the Destination UID of the RDMCommand */
+  const UID& DestinationUID() const { return m_destination; }
+
+  /** @brief Returns the Transaction Number of the RDMCommand */
+  uint8_t TransactionNumber() const { return m_transaction_number; }
+
+  /** @brief Returns the Message Count of the RDMCommand */
+  uint8_t MessageCount() const { return m_message_count; }
+
+  /** @brief Returns the SubDevice of the RDMCommand */
+  uint16_t SubDevice() const { return m_sub_device; }
+
+  /** @brief Returns the Port ID of the RDMCommand */
+  uint8_t PortIdResponseType() const { return m_port_id; }
+
+  /** @brief Returns the Parameter ID of the RDMCommand */
+  uint16_t ParamId() const { return m_param_id; }
+
+  /** @brief Returns the Parameter Data of the RDMCommand */
+  const uint8_t *ParamData() const { return m_data; }
+
+  /** @brief Returns the Size of the Parameter Data of the RDMCommand */
+  unsigned int ParamDataSize() const { return m_data_length; }
+
+  /** @} */
+
+  /**
+   * @brief Used to print the data in an RDM Command to a CommandPrinter
+   * @param printer CommandPrinter which will use the information
+   * @param summarize enable a one line summary
+   * @param unpack_param_data if the summary isn't enabled, this controls if
```

Comment:
The options are just passed through to the CommandPrinter. I'll clean this up later and turn them into a struct.